### DISCUSSION TITLE
Determine the current save_path of a torrent in a workflow

### DISCUSSION THREAD

**Discussion by tlaerm** [porla/porla#290] [Q&A] [answered]
Hi everyone, 
I am trying to write a workflow that moves finished torrents. It has to look at the path the torrent is currently saved at. This seems to be available at ```save_path``` somewhere.
The example workflow looks like this:
```
local workflows = require("workflows")
local torrent   = require("workflows.actions.torrent")
local added     = require("workflows.triggers.torrent").added

function porla.init()
    workflows.add({
        trigger = added(),
        actions = {
            torrent.move({
                path = "/tmp"
            })
        }
    })
end
```
I have tried accessing torrent:status() or directly torrent.save_path but I seem to be on the wrong track. Any help would be greatly appreciated. (Please be gentle, I am a Lua noob.)

**Comment by vktr** [comment, marked as answer]
Hi! To move a torrent when it finishes, with a check on its current save path (or other information), you should be able to do something like this,

```lua
local workflows = require("workflows")
local torrent   = require("workflows.actions.torrent")
local finished  = require("workflows.triggers.torrent").finished

function porla.init()
    workflows.add({
        trigger = finished(),
        actions = {
            torrent.move({
                path = function(ctx)
                    local status = ctx.torrent:status()
                    -- compare status.save_path to something
                    return "/new/save/path"
                end
            })
        }
    })
end
```

Refer to https://porla.org/plugins/packages/sessions#torrentstatus to see all properties available on the `TorrentStatus` object.

**Reply by tlaerm** [reply]
Got it working thanks to your example. Thank you. Now the files get moved to the correct location and I can even use regex to make this work even better.
